services/src/datasets/external/netcdfcf/overviews.rs

```diff
@@ -292,156 +405,335 @@ fn store_metadata(
+    compression_format: String,
+    compression_level: String,
+    num_threads: String,
+    resampling_method: String,
+}
+
+impl CogRasterCreationOptions {
+    fn new(resampling_method: Option<ResamplingMethod>) -> Result<Self> {
+        const COMPRESSION_FORMAT: &str = "LZW"; // this is the GDAL default
+        const DEFAULT_COMPRESSION_LEVEL: u8 = 6; // this is the GDAL default
+        const DEFAULT_RESAMPLING_METHOD: ResamplingMethod = ResamplingMethod::Nearest;
+
+        let gdal_options = get_config_element::<crate::util::config::Gdal>()
+            .boxed_context(error::CannotCreateOverviews)?;
+        let num_threads = gdal_options.compression_num_threads.to_string();
+        let compression_format = gdal_options
+            .compression_algorithm
+            .as_deref()
+            .unwrap_or(COMPRESSION_FORMAT)
+            .to_string();
+        let compression_level = gdal_options
+            .compression_z_level
+            .unwrap_or(DEFAULT_COMPRESSION_LEVEL)
+            .to_string();
+        let resampling_method = resampling_method
+            .unwrap_or(DEFAULT_RESAMPLING_METHOD)
+            .to_string();
+
+        Ok(Self {
+            compression_format,
+            compression_level,
+            num_threads,
+            resampling_method,
+        })
+    }
+}
+
+impl CogRasterCreationOptions {
+    fn options(&self) -> Vec<RasterCreationOption<'_>> {
+        const COG_BLOCK_SIZE: &str = "512";
+
+        vec![
+            RasterCreationOption {
+                key: "COMPRESS",
+                value: &self.compression_format,
+            },
+            RasterCreationOption {
+                key: "LEVEL",
+                value: &self.compression_level,
+            },
+            RasterCreationOption {
+                key: "NUM_THREADS",
+                value: &self.num_threads,
+            },
+            RasterCreationOption {
+                key: "BLOCKSIZE",
+                value: COG_BLOCK_SIZE,
+            },
+            RasterCreationOption {
+                key: "BIGTIFF",
+                value: "IF_SAFER", // TODO: test if this suffices
+            },
+            RasterCreationOption {
+                key: "RESAMPLING",
+                value: &self.resampling_method,
+            },
+        ]
+    }
+}
+
 fn generate_loading_info(
     dataset: &Dataset,
     overview_dataset_path: &Path,
     time_coverage: &TimeCoverage,
-) -> Result<Metadata> {
-    const SAMPLE_BAND: usize = 1;
+) -> Result<MetaDataDefinition> {
+    const ONLY_BAND: usize = 1;
 
     let result_descriptor = raster_descriptor_from_dataset(dataset, 1, None)
         .boxed_context(error::CannotGenerateLoadingInfo)?;
 
-    let params = gdal_parameters_from_dataset(
+    let mut params = gdal_parameters_from_dataset(
         dataset,
-        SAMPLE_BAND,
+        ONLY_BAND,
         overview_dataset_path,
-        Some(0), // we calculate offsets in our source
+        Some(ONLY_BAND),
```

Comment:
why is this 1 now, when before, it was 0?

Reply:
GDAL bands start with one and we use always the first band.
0, previously, was invalid, but we adjusted it during the load process.